activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -506,7 +510,11 @@ def copy_table_contents(from, to, columns, rename = {}) #:nodoc:
         end
 
         def sqlite_version
-          @sqlite_version ||= SQLite3Adapter::Version.new(select_value('select sqlite_version(*)'))
+          @sqlite_version ||= SQLite3Adapter::Version.new(version_string)
+        end
+
+        def version_string
+          @version_string ||= select_value('select sqlite_version(*)')
         end
```

Comment:
-1 on an additional memoized method. We only need to stub `sqlite_version` and test that it affects `#supports_multi_insert?`.